core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -232,6 +232,11 @@ class HadoopRDD[K, V](
         logWarning(s"${jobConf.get(FileInputFormat.INPUT_DIR)} doesn't exist and no" +
             s" partitions returned from this path.", e)
         Array.empty[Partition]
+      case e: IOException if e.getMessage.contains("Not a file") =>
+        val path = e.getMessage.split(":").map(_.trim).apply(2)
+        throw new IOException(s"Path: ${path} is a directory, it is not" +
+          s" allowed for `serde` reader when" +
```

Comment:
The code path here is in core module. The reader may not be `serde` reader.

Reply:
> The code path here is in core module. The reader may not be `serde` reader.

How about current?

Reply:
How about:
```
s"Path: ${path} is a directory, which is not supported by the record reader when `mapreduce.input.fileinputformat.input.dir.recursive` is false." 
```

Reply:
> How about:
> 
> ```
> s"Path: ${path} is a directory, which is not supported by the record reader when `mapreduce.input.fileinputformat.input.dir.recursive` is false." 
> ```

updated